x-pack/legacy/plugins/maps/public/actions/map_actions.js

```diff
@@ -663,13 +663,39 @@ export function updateSourceProp(layerId, propName, value, newLayerType) {
       layerId,
       propName,
       value,
-      newLayerType,
     });
+    if (newLayerType) {
+      dispatch(updateLayerType(layerId, newLayerType));
+    }
     await dispatch(clearMissingStyleProperties(layerId));
     dispatch(syncDataForLayer(layerId));
   };
 }
 
+function updateLayerType(layerId, newLayerType) {
+  return (dispatch, getState) => {
+    const layer = getLayerById(layerId, getState());
+    if (!layer || layer.getType() === newLayerType) {
+      return;
+    }
+    layer.getInFlightRequestTokens().forEach(requestToken => {
+      dispatch(cancelRequest(requestToken));
+    });
+    dispatch({
+      type: UPDATE_LAYER_PROP,
+      id: layerId,
+      propName: '__dataRequests',
+      newValue: [],
+    });
```

Comment:
Should we pull this logic out into a separate wrapper action and just call it from here? There are a couple of places where we're cancelling requests. Seems like we'd want to reset `__dataRequests` for these as well

Reply:
getInFlightRequestTokens is also called in:
1) removeLayerFromLayerList - No need to clear data requests since the layer is getting removed from the layer list
2) cancelAllInFlightRequests - This called when the map component unmounts. It would not hurt to clear data requests here. I can update.